charts/drax/values.yaml

```
@@ -819,6 +835,45 @@ influxdb:
         CREATE DATABASE "db_5G" WITH DURATION 5d REPLICATION 1 NAME "rp_5d_5G"
 
 
+influxdb2:
+  enabled: true
+
+  podLabels:
+    drax/role: ric
+    drax/component-name: influxdb2
+
+  adminUser:
+    organization: "accelleran"
+    bucket: "default"
+    retention_policy: "0s"
+    user: "admin"
+    password: ""
+    token: "accelleran"
```

Insecure.

We can just let influxdb generate a random token as it does that by default.
Everything can be setup to use this secret (and grafana for example already is as part of this PR). In telemetry collector an environment variable was also added to be able to do this so there shouldn't be anything blocking us from doing the correct thing here.

```suggestion
    token: ""
```

---

I believe we discussed this problem, where the persistence gets involved with the incorrect token being picked up after a re-install, so we agreed to inject the token here instead?

---

We shouldn't uninstall and install already due to the DU influxdb instance. So not a problem in my opinion as it's already something we need to take into account. In general we should be doing `upgrade`s anyway.

The "solution" in case the release is removed (and the token is lost) would be to just remove the persistence volume claim so we start from scratch (at the cost of data loss of course, but once the token is already lost the data is also lost).

---

I agree that this is insecure but I think at the moment not hard-coding the secret somewhere will cause more trouble than good. The problem is that most people (also external clients) use helm install/uninstall. Wouldn't we also have the same issue we powercycle the server etc? The workaround of removing the pvc means that we will lose all our data. This is indeed part of a bigger AAA story. Is it worth for us to create a drax-level "hardcoded" secret that will be injected to influxdb for now? And then as part of the AAA we can see how to inject it in a more secure way?

---

A reboot won't have the same issue as all kubernetes resources are stored persistenly in etcd, thus the randomly generated token is still available, thus the storage for influxdb can also still be used.

If we don't do this now, then we will in the future require a breaking change with data loss, which we can avoid by just already using it.

One thing we can maybe do as the influx helm chart doesn't allow it is basically generate the secret ourselves with the annotation `helm.sh/resource-policy: keep`. Then it won't be removed after an uninstall. (https://helm.sh/docs/howto/charts_tips_and_tricks/#tell-helm-not-to-uninstall-a-resource)

---

The last point of course has the problem of not having a `tpl` call for `.Values.influxdb2.adminUser.existingSecret`, so not possible without upstream changes then...